src/performance.js

```diff
@@ -58,8 +66,48 @@ export class Performance {
 
     /** @const @private {!Array<TickEventDef>} */
     this.events_ = [];
+
+    /** @private {boolean} */
+    this.isViewerVisible_ = false;
+
+    this.win.addEventListener('unload', this.flush.bind(this));
   }
 
+  /**
+   * Gets primary amp core services.
+   * @return {!Performance}
+   */
+  init() {
```

Comment:
I see. Definitely if you need them nullable - than this is all ok. Although then I'd rename `init` into something that's much clearer, such as "runtimeAvailable".